### PR TITLE
fixes #7 #9 #10 #11 added OrderedElements, AnyN, and Less

### LINKED ISSUE #7: Implement Set.ElementsBy
Like `Elements` but uses `RangeBy`.

### PR DESCRIPTION
**Pull request by nofun97**
Fixes #7 #9 #10  #11 

Added `ElementsBy`, `AnyN` and `Less` data structure

### CODE REVIEW

File: set.go [hunk 1]
```diff
@@ -61,6 +70,22 @@ func (s Set) Any() interface{} {
 	panic("empty set")
 }
 
+// AnyN returns a set N arbitrary element from the Set
```
Comment (marcelocantos) [inline set.go]: `... set of ... elements ...`.
`s/$/./`

File: set.go [hunk 2]
```diff
@@ -53,6 +53,15 @@ func (s Set) Elements() []interface{} {
 	return result
 }
 
+// Takes elements in a defined order
```
Comment (marcelocantos) [inline set.go]: `s/$/./`

File: set.go [hunk 3]
```diff
@@ -61,6 +70,22 @@ func (s Set) Any() interface{} {
 	panic("empty set")
 }
 
+// AnyN returns a set N arbitrary element from the Set
+func (s Set) AnyN(n int) Set {
+	if s.Count() == 0 {
+		if n == 0 {
```
Comment (marcelocantos) [inline set.go]: Test not required. It's valid to ask for more elements than the set has.

File: set.go [hunk 4]
```diff
@@ -61,6 +70,22 @@ func (s Set) Any() interface{} {
...
+			return NewSet()
+		}
+		panic("empty set")
+	}
+	count := 0
+	setBuilder := NewSetBuilder(0)
```
Comment (marcelocantos) [inline set.go]:
```go
var setBuilder SetBuilder
```

File: node_iter.go [hunk 5]
```diff
@@ -2,6 +2,9 @@ package frozen
 
 import "container/heap"
 
+// Less dictates the order of two elements
```
Comment (marcelocantos) [inline node_iter.go]: `s/$/./`

File: set_test.go [hunk 6]
```diff
@@ -137,6 +137,33 @@ func TestSetAny(t *testing.T) {
...
+
+func TestSetElementsBy(t *testing.T) {
+	t.Parallel()
+
+	s := NewSet(5, 1, 2, 4, 3)
+	less := func(a, b interface{}) bool { return a.(int) < b.(int) }
+	assert.Equal(t, []interface{}{1, 2, 3, 4, 5}, s.ElementsBy(less))
+
+	less = func(a, b interface{}) bool { return a.(int) > b.(int) }
+	assert.Equal(t, []interface{}{5, 4, 3, 2, 1}, s.ElementsBy(less))
```
Comment (marcelocantos) [inline set_test.go]: Try a much larger input. Use `Iota`.

File: set_test.go [hunk 7]
```diff
@@ -137,6 +137,33 @@ func TestSetAny(t *testing.T) {
...
+	assert.Panics(t, func() { s.AnyN(1) })
+	assert.Equal(t, 0, s.AnyN(0).Count())
+
+	s = s.With(1)
+	assert.Equal(t, 1, s.AnyN(1).Count())
+	assert.Equal(t, 1, s.AnyN(2).Count())
+
+	s = s.With(2)
+	assert.Equal(t, 1, s.AnyN(1).Count())
+	assert.True(t, s.AnyN(2).Equal(NewSet(1, 2)))
```
Comment (marcelocantos) [inline set_test.go]: Try much larger values of N.

**Review by marcelocantos — CHANGES_REQUESTED** [review]

File: set.go (hunk 2 above)
Comment (nofun97) [reply set.go]: sorry what does `s/$/./` mean?

File: set_test.go (hunk 7 above)
Comment (nofun97) [reply set_test.go]: added

File: set_test.go (hunk 6 above)
Comment (nofun97) [reply set_test.go]: changes added

File: set.go (hunk 4 above)
Comment (nofun97) [reply set.go]: changed to that

File: set.go (hunk 3 above)
Comment (nofun97) [reply set.go]: removed the test

File: set.go (hunk 1 above)
Comment (nofun97) [reply set.go]: added

File: set.go (hunk 2 above)
Comment (nofun97) [reply set.go]: added

File: node_iter.go (hunk 5 above)
Comment (nofun97) [reply node_iter.go]: added

File: set_test.go [hunk 8]
```diff
@@ -137,6 +137,33 @@ func TestSetAny(t *testing.T) {
...
+	s = Iota(1<<12 - 1)
+	assert.Equal(t, 1, s.AnyN(1).Count())
+	assert.True(t, s.AnyN(1<<12-1).Equal(Iota(1<<12-1)))
+}
+
+func TestSetOrderedElements(t *testing.T) {
+	t.Parallel()
+
+	s := Iota(1<<12 - 1)
+	less := func(a, b interface{}) bool { return a.(int) < b.(int) }
```
Comment (ChloePlanet) [inline set_test.go]: less := Less(func ...)

File: node_iter.go [hunk 9]
```diff
@@ -2,6 +2,9 @@ package frozen
 
 import "container/heap"
 
+// Less dictates the order of two elements.
+type Less func(a, b interface{}) bool
```
Comment (ChloePlanet) [inline node_iter.go:6]: L55: replace `func(a, b interface{}) bool` with `Less`

**Review by ChloePlanet — APPROVED** [review]

**Pull request merged**